index.Rmd

```diff
@@ -1,74 +1,545 @@
 
 See [http://rmarkdown.rstudio.com/](http://rmarkdown.rstudio.com/) for all the amazing things you can do.
 
 
 Here's my first code chunk.
-```{r}
-1+2
+```{r,message=F, warning=F}
+library(tidycensus)
+library(sf)
+library(tidyverse)
+library(haven)
+library(viridis)
+library(leaflet)
+library(tidyr)
+library(kableExtra)
+
+# census_api_key ("7e9b9dfdae5d56fbf2a5c8e243c74bccbe74af98", install=TRUE,overwrite=TRUE)
+
+census_api_key(Sys.getenv("CENSUS_API_KEY"),install=TRUE, overwrite = TRUE)
+readRenviron("~/.Renviron")
+
 ```
 
-Load any required packages in a code chunk (you may need to install some packages):
 
-```{r, message=F, warning=F}
+```{r, message=F, warning=F,results='hide'}
 library(tidyverse)
 library(leaflet)
 library(kableExtra)
 knitr::opts_chunk$set(cache=TRUE)  # cache the results for quick compiling
 ```
 
 ## Download and clean all required data
-```{r}
-n=20
-data=data.frame(x=runif(n,-180,180),
-                y=runif(n,-60,60),
-                size = runif(n, 5, 20),
-                category = factor(
-                  sample(letters[1:5], n, replace = TRUE)
-                  ),
-                value = rnorm(n))
+```{r,message=F, warning=F}
+year=2012:2019
+tract.list=NULL
+county.list=NULL
+tract_county_merge.list=NULL
+for (i in 1:8){
+  d1= get_acs(geography = "tract", state = "NY",
+              county = "Erie", variables = c(TotalP_t="B01001_001",
+                TotalRace_t= "B02001_001",
+                                             White_t="B02001_002",
+                                             nonHispanicWhite_t="B03002_003",
+                                             nonHispanicWhiteTotal_t="B03002_001",
+                                             Homeowner_t="B25003_002",
+                                             renterHS_t="B25003_003",
+                                             TenureHSTotal_t="B25003_001",
+                                             collegedegree_t="B15003_022",
+                                             degreeTotal_t="B15003_001",
+                                             # HSIncome_t=c("B19001_001","B19001_002","B19001_003","B19001_004","B19001_005","B19001_006","B19001_007","B19001_008","B19001_009"),##how to do it in a quicker way??
+                                             medianHSincome_t="B19013_001",
+                                             medianHValue_t="B25077_001",
+                                             medianGRent_t="B25064_001",
+                                             Poverty_t="B17001_002",
+                                             PovertyTotal_t="B17001_001",
+                                             ForeignBorn_t="B05002_009",
+                                             NavitityTotal_t="B05002_001",
+                                             Profjobs_t="C24070_010",
+                                             jobsTotal_t="C24070_001"
+              ),year=year[i],geometry = TRUE,
```

Comment:
can you give it multiple years at the same time?  Just give it a vector of years here?  I'm not sure, but I thought so.